### PR TITLE
[FSSDK-9090] Default instantiation of OdpManager for Optimizely constructors

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Fix OdpManager not having a default instantiation in `Optimizely` client

`OptimizelyFactory` provides a default which is desirable.  The `Optimizely` constructors should follow this.

## Test plan

- Add covering and passing unit test for `Optimizely`
- FSC integration tests should continue to pass

## Issues
- FSSDK-9090

### CODE REVIEW

File: OptimizelySDK/Optimizely.cs [hunk 1]
```diff
@@ -438,8 +442,8 @@ private bool ValidateInputs(string datafile, bool skipJsonValidation)
             userAttributes = userAttributes ?? new UserAttributes();
 
             var userContext = CreateUserContextCopy(userId, userAttributes);
-            var variation = DecisionService.GetVariation(experiment, userContext, config)?.
-                ResultObject;
+            var variation = DecisionService.GetVariation(experiment, userContext, config)
+                ?.ResultObject;
```
Comment (andrewleap-optimizely) [inline OptimizelySDK/Optimizely.cs:446]: I like the use of the leading period in method chaining- what do you think about leaving the null conditional operator on the previous line, grouping it with its relevant method?

**Review by andrewleap-optimizely — APPROVED** [review]
LGTM (with small nit)

File: OptimizelySDK/Optimizely.cs (hunk 1 above)
Comment (andrewleap-optimizely) [reply OptimizelySDK/Optimizely.cs:446]: 🤷‍♂️ you win some, you lose some

**Review by mnoman09 — CHANGES_REQUESTED** [review]
Overall changes lgtm. But how are we going to disable the odp? Previously we assumed that if user is not passing odpManager then it will be considered as disabled. In my opinion if someone is using OptimizelyFactory to initialize Optimizely then odpManager should be enabled by default otherwise do not initialize it and consider it disabled. 
@jaeopt What do you suggest?

**Comment by jaeopt** [comment]
> I have a question. How are we going to disable the odp? Previously we assumed that if user is not passing odpManager then it will be considered as disabled. In my opinion if someone is using OptimizelyFactory to initialize Optimizely then odpManager should be enabled by default otherwise do not initialize it and consider it disabled. @jaeopt What do you suggest?

ODP will be enabled unless they

**Comment by mikechu-optimizely** [comment]
@mnoman09 I updated the https://github.com/optimizely/csharp-testapp/pull/83

**Comment by mikechu-optimizely** [comment]
Is it ok to have this PR open @jaeopt given my change to the test app to support FSC?

**Comment by jaeopt** [comment]
> Is it ok to have this PR open @jaeopt given my change to the test app to support FSC?

@mikechu-optimizely oops, did I close the PR? I was wondering my comment has been broken and not realized I closed it by mistake. Sorry :)

**Comment by mikechu-optimizely** [comment]
Sweet. Passing now.  I just hope I satisifed @mnoman09 's question to get a LGTM

**Comment by mikechu-optimizely** [comment]
Hey @mnoman09, I think you might be out, but I'm going to go ahead and merge this change.  DM me if you wanna talk more about it.  I'm happy to open more PRs to get it right.

**Pull request merged**